index.html

```diff
@@ -1239,37 +1257,57 @@ <h2>
             `"hidden"`, return [=a promise rejected with=] an
             "{{InvalidStateError}}" {{DOMException}}.
             </li>
-            <li>Let |resetResultPromise:Promise| be [=a new promise=].
+            <li>Increment [=this=].{{GamepadHapticActuator/[[sequenceId]]}}.
             </li>
             <li>If [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-            is not `null`, do the following steps [=in parallel=]:
+            is not `null`:
               <ol>
                 <li>Let |effectPromise| be
                 [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}.
                 </li>
+                <li>Set
+                [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}} to
+                `null`.
+                </li>
+                <li>[=Queue a global task=] on the [=relevant global object=]
+                of [=this=] using the [=gamepad task source=] to [=resolve=]
+                |effectPromise| with {{GamepadHapticsResult/"preempted"}}.
+                </li>
+              </ol>
+            </li>
+            <li>Let [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
+            be [=a new promise=].
+            </li>
+            <li>Let |resetSequenceId:unsigned long| be
+            [=this=].{{GamepadHapticActuator/[[sequenceId]]}}.
+            </li>
+            <li>Do the following steps [=in parallel=]:
+              <ol>
                 <li>[=Stop haptic effects=] on [=this=]'s gamepad's actuator.
                 </li>
-                <li>If the effect has been successfully stopped, do:
+                <li>If the effect has been successfully stopped, [=queue a
+                global task=] on the [=relevant global object=] of [=this=]
+                using the [=gamepad task source=] to run the following steps:
                   <ol>
-                    <li>If |effectPromise| and
-                    [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-                    are still the same, set
-                    [=this=].{{GamepadHapticActuator/[[playingEffectPromise]]}}
-                    to `null`.
-                    </li>
-                    <li>[=Queue a global task=] on the [=relevant global
-                    object=] of [=this=] using the [=gamepad task source=] to
-                    [=resolve=] |effectPromise| with
-                    {{GamepadHapticsResult/"preempted"}}.
+                    <li>If [=this=].{{GamepadHapticActuator/[[sequenceId]]}} is
+                    |resetSequenceId|:
```

Comment:
same re early return.

Reply:
On second thought, not sure if an early return makes sense, given we always still need to return the promise at the end(we'd end up with two return statements).

Reply:
Done